notebook/tests/notebook/execute_code.js

```diff
@@ -1,115 +0,0 @@
-        this.test.assertEquals(result.text, '10\n', 'cell execute (using js)');
-    });
-
-
-    // do it again with the keyboard shortcut
-    this.thenEvaluate(function () {
-        var cell = IPython.notebook.get_cell(0);
-        cell.set_text('a=11; print(a)');
-        cell.clear_output();
-    });
-
-    this.then(function(){
-        
-        this.trigger_keydown('shift-enter');
-    });
-
-    this.wait_for_output(0);
-
-    this.then(function () {
-        var result = this.get_output_cell(0);
-        var num_cells = this.get_cells_length();
-        this.test.assertEquals(result.text, '11\n', 'cell execute (using ctrl-enter)');
-        this.test.assertEquals(num_cells, 2, 'shift-enter adds a new cell at the bottom')
-    });
-    
-    // do it again with the keyboard shortcut
-    this.thenEvaluate(function () {
-        IPython.notebook.select(1);
-        IPython.notebook.delete_cell();
-        var cell = IPython.notebook.get_cell(0);
-        cell.set_text('a=12; print(a)');
-        cell.clear_output();
-    });
-
-    this.then(function(){
-        this.trigger_keydown('ctrl-enter');
-    });
-
-    this.wait_for_output(0);
-
-    this.then(function () {
-        var result = this.get_output_cell(0);
-        var num_cells = this.get_cells_length();
-        this.test.assertEquals(result.text, '12\n', 'cell execute (using shift-enter)');
-        this.test.assertEquals(num_cells, 1, 'ctrl-enter adds no new cell at the bottom')
-    });
-
-    // press the "play" triangle button in the toolbar
-    this.thenEvaluate(function () {
-        var cell = IPython.notebook.get_cell(0);
-        IPython.notebook.select(0);
-        cell.clear_output();
-        cell.set_text('a=13; print(a)');
-        $("button[data-jupyter-action='jupyter-notebook:run-cell-and-select-next']")[0].click()
-    });
-    
-    this.wait_for_output(0);
-
-    this.then(function () {
-        var result = this.get_output_cell(0);
-        this.test.assertEquals(result.text, '13\n', 'cell execute (using "play" toolbar button)')
-    });
-
-    // run code with skip_exception
-    this.thenEvaluate(function () {
-        var cell0 = IPython.notebook.get_cell(0);
-        cell0.set_text('raise IOError');
-        IPython.notebook.insert_cell_below('code',0);
-        var cell1 = IPython.notebook.get_cell(1);
-        cell1.set_text('a=14; print(a)');
-        cell0.execute(false);
-        cell1.execute();
-    });
-
-    this.wait_for_output(1);
-
-    this.then(function () {
-        var result = this.get_output_cell(1);
-        this.test.assertEquals(result.text, '14\n', "cell execute, don't stop on error");
-    });
-
-    this.thenEvaluate(function () {
-        var cell0 = IPython.notebook.get_cell(0);
-        cell0.set_text('raise IOError');
-        IPython.notebook.insert_cell_below('code',0);
-        var cell1 = IPython.notebook.get_cell(1);
-        cell1.set_text('a=14; print(a)');
-        cell0.execute();
```

Comment:
@takluyver Hi, I was struggling to find a way to ignore the error cell, so I read this test again and notice that only the cell 1 was being executed, I replicated this with python and if you remove the `cell0.execute(false)` line, the test passes anyway
Do you know if there's any real way to ignore a cell? I couldn't find it in the UI neither

Reply:
I don't think this is exposed in the UI at all, but this is the idea of the test:

Normally, if you select two cells together and run them and the first one raises an exception, the execution request for the second one will be cancelled. That's what this block (from line 97) is testing.

You can override this behaviour in the Javascript API - if you call `cell.execute(false)` and it errors, later cells queued for execution will still be executed. The block above (lines 80 - 95) is testing this. So if you remove the `cell0.execute(false)` line,  it will definitely pass, but it's missing what it's designed to test.

I think the only way to test this from Python will be to execute some Javascript to get those cell objects and call their `.execute()` method.